### PR TITLE
Add other field to course length

### PR DESCRIPTION
**Pull request by MartinBentley**
### Context
In the Course Fees/Salary enrichment views the "Other" option did not show a text box
### Guidance to review
courseenrichment
preview

### CODE REVIEW

File: src/ui/FeatureFlags.cs [hunk 1]
```diff
@@ -22,6 +22,6 @@ public FeatureFlags(IConfigurationSection config)
 
         public bool ShowCourseLiveView => ShouldShow(FEATURE_COURSE_LIVEVIEW);
 
-        private bool ShouldShow(string key) => _config.GetValue(key, false);
+        private bool ShouldShow(string key) => _config.GetValue(key, true);
```
Comment (dankmitchell) [inline src/ui/FeatureFlags.cs]: Need to set this back to `false`

File: src/ui/ViewModels/CourseFeesEnrichmentViewModel.cs [hunk 2]
```diff
@@ -51,8 +57,16 @@ public void CopyFrom(CourseEnrichmentModel model)
             {
                 CourseLength = courseLength;
             }
-
-            if(model.FeeUkEu.HasValue)
+            //if (!CourseLength.HasValue && !string.IsNullOrEmpty(model.CourseLength))
```
Comment (dankmitchell) [inline src/ui/ViewModels/CourseFeesEnrichmentViewModel.cs]: No need to keep commented out code

File: src/ui/ViewModels/CourseSalaryEnrichmentViewModel.cs [hunk 3]
```diff
@@ -24,8 +25,12 @@ public class CourseSalaryEnrichmentViewModel : ICourseEnrichmentViewModel
         public void MapInto(ref CourseEnrichmentModel enrichmentModel)
         {
             var courseLength = CourseLength.HasValue ? CourseLength.Value.ToString() : null;
+            enrichmentModel.CourseLengthOther = courseLength == "Other" ?  CourseLengthOther : null;//remove this line for a single field
+
+            //enrichmentModel.CourseLength = courseLength == "Other" ? (string.IsNullOrEmpty(CourseLengthOther) ? null : CourseLengthOther) : courseLength;
```
Comment (dankmitchell) [inline src/ui/ViewModels/CourseSalaryEnrichmentViewModel.cs]: Do we need this line?

File: src/ui/ViewModels/CourseSalaryEnrichmentViewModel.cs [hunk 4]
```diff
@@ -40,8 +45,16 @@ public void CopyFrom(CourseEnrichmentModel model)
             {
                 CourseLength = courseLength;
             }
-            
-            if(!string.IsNullOrEmpty(model.SalaryDetails))
+            //if (!CourseLength.HasValue && !string.IsNullOrEmpty(model.CourseLength))
```
Comment (dankmitchell) [inline src/ui/ViewModels/CourseSalaryEnrichmentViewModel.cs]: Do we need this commented out code?

File: src/ui/FeatureFlags.cs (hunk 1 above)
Comment (MartinBentley) [reply src/ui/FeatureFlags.cs]: Oops!

File: src/ui/Services/CourseMapper.cs [hunk 5]
```diff
@@ -44,16 +44,16 @@ public SearchAndCompare.Domain.Models.Course MapToSearchAndCompareCourse(ApiClie
                 Duration = MapCourseLength(courseEnrichmentModel.CourseLength, courseEnrichmentModel.CourseLengthOther),
```
Comment (dankmitchell) [inline src/ui/Services/CourseMapper.cs]: Alot of white space in this file...

File: ManageCoursesUi.Tests/Services/CourseMapperTests.cs [hunk 6]
```diff
@@ -90,7 +90,7 @@ public void MapToSearchAndCompareCourse()
                 }
             );
 
-            res.Duration.Should().Be("1 year");
+            res.Duration.Should().Be("One year");
```
Comment (dankmitchell) [inline ManageCoursesUi.Tests/Services/CourseMapperTests.cs]: This should be `1` not `One`

File: src/ui/Services/CourseMapper.cs [hunk 7]
```diff
@@ -180,10 +180,11 @@ public SearchAndCompare.Domain.Models.Course MapToSearchAndCompareCourse(ApiClie
             return mappedCourse;
         }
 
-        private string MapCourseLength(string courseLength)
+        private string MapCourseLength(string courseLength, string courseLengthOther)
         {
-            return courseLength == "OneYear" ? "1 year"
-                : courseLength == "TwoYears" ? "Up to 2 years"
+            return courseLength == "OneYear" ? "One year"
```
Comment (dankmitchell) [inline src/ui/Services/CourseMapper.cs]: This should be `1 year` not `One year`

File: src/ui/Services/CourseMapper.cs [hunk 8]
```diff
@@ -180,10 +180,11 @@ public SearchAndCompare.Domain.Models.Course MapToSearchAndCompareCourse(ApiClie
...
+                : courseLength == "TwoYears" ? "Up to two years"
```
Comment (dankmitchell) [inline src/ui/Services/CourseMapper.cs]: This should be `Up to 2 years` not `Up to two years`

**Comment by defong** [comment]
Is it me or is this a bit to complex?


Kinda expected view model to have a string property only

viewModel.CourseLength = null;// if apiModel.CourseLength
viewModel.CourseLength = "one year" //if apiModel.CourseLength == CourseLength.OneYear
viewModel.CourseLength = "two year" //if apiModel.CourseLength == CourseLength.TwoYear
viewModel.CourseLength = "others" //if (apiModel.CourseLength != CourseLength.TwoYear && apiModel.CourseLength == CourseLength.OneYear)

File: src/ui/Services/CourseMapper.cs [hunk 9]
```diff
@@ -44,16 +44,16 @@ public SearchAndCompare.Domain.Models.Course MapToSearchAndCompareCourse(ApiClie
                 Duration = MapCourseLength(courseEnrichmentModel.CourseLength),
                 Name = ucasCourseData.Name,
                 ProgrammeCode = ucasCourseData.CourseCode,
-                Provider = provider,
-                ProviderCodeName = ucasInstData.InstBig, // ???
+                Provider = provider,                
+                ProviderCodeName = ucasInstData.InstBig, // ???
```
Comment (timabell) [inline src/ui/Services/CourseMapper.cs]: added trailing whitespace, reformat the file

Comment (defong) [reply src/ui/Services/CourseMapper.cs]: whats `// ???`

Comment (MartinBentley) [reply src/ui/Services/CourseMapper.cs]: That's been in there since the file was first created. It doesn't mean anything so I'll remove it

**Pull request merged**